website/plugins/proto-messages/index.js

```diff
@@ -211,3 +260,46 @@ function getMatchingType(sourceDictionary, line) {
 
   return match;
 }
+
+const TLD = "(\\.\\w+)+";
+const LINK_ONLY = `https?\:\/\/.+${TLD}`;
+const LINK_WITH_TEXT = `\\[.+\\]\\(${LINK_ONLY}\\)`;
+
+const LINK_WITH_TEXT_SEPARATOR = "](";
+
+function getLinksFromALine(line) {
+  // Matches [any](any) or any://any.
+  const lineRegex = new RegExp(`${LINK_ONLY}|${LINK_WITH_TEXT}`, "g");
```

Comment:
optional optimization: to initialize this outside the function.

Reply:
Last I tried it led to the regex not matching anything due to the `g` flag (for the N-th `.exec()`), but yeah I think it was my mistake. We already have the `while (match)` which would cover that case. Let me see what I can do.

Great point!

Reply:
Cool! I moved it outside and yeah it works as intended still. Thanks for the feedback!